primitives/src/ethereum_checked.rs

```diff
@@ -45,7 +47,7 @@
     /// Amount to transfer.
     pub value: U256,
     /// Input of a contract call.
-    pub input: BoundedVec<u8, ConstU32<MAX_ETHEREUM_TX_INPUT_SIZE>>,
+    pub input: EthereumTxInput,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change to use the `EthereumTxInput` type in the `CheckedEthereumTx` struct is appropriate and aligns with the goal of modularity. Make sure that the initialization and handling of the `input` field are consistent with the use of this new type. Verify that the maximum input size is correctly enforced and that there are no vulnerabilities related to input overflows or invalid data sizes.

This change seems well-structured and appropriately modular. However, I recommend thoroughly testing these changes in the context of Ethereum transaction processing to ensure the correctness and security of the code. LGTM!


<!-- This is an auto-generated comment by OSS CodeRabbit -->